kythe/cxx/common/protobuf_metadata_file.cc

```diff
@@ -36,6 +36,9 @@ proto::VName ProtobufMetadataSupport::VNameForAnnotation(
     out.set_corpus(context_vname.corpus());
   }
   out.set_path(annotation.source_file());
+  if (absl::EndsWith(annotation.source_file(), ".gen.proto")) {
+    out.set_root(context_vname.root());
+  }
```

Comment:
I'm hesitant to hard-code more file extension-specific behavior in the indexer.  Regardless, this should come in the same block as setting the corpus (as in, it should only happen if looking up the annotation-provided source file VName directly fails).

Reply:
Notably, because `context_vname` is generated from the name of the metadata file, which will always be "generated", it's possible (though unlikely, currently) that the root of the generated proto will differ from the root used here.

Reply:
Moved set_root to the same block as setting the corpus.